src/Data/Text/Builder/Linear/Core.hs

```diff
@@ -246,6 +248,57 @@ appendBounded maxSrcLen appender (Buffer (Text dst dstOff dstLen)) = Buffer $ ru
   pure $ Text new dstOff (dstLen + srcLen)
 {-# INLINE appendBounded #-}
 
+-- | Low-level routine to append data of unknown size to a 'Buffer', giving
+-- the action the choice between two strategies.
+--
+-- See also: 'appendBounded'.
+appendBounded'
```

Comment:
_Internal_ new helper. If deemed a useful addition to the API, maybe we need a better name.

Reply:
`Data.Text.Builder.Linear.Core` is a public module, so it's not that internal. Maybe we can put the helper somewhere else to avoid nameshedding?